index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestIndexRepair.scala

```diff
@@ -119,6 +119,19 @@ class TestIndexRepair extends QueryTest with BeforeAndAfterAll {
     sql("drop table if exists maintable")
   }
 
+  test("reindex command with stale files") {
+    sql("drop table if exists maintable")
+    sql("CREATE TABLE maintable(a INT, b STRING, c STRING) stored as carbondata")
+    sql("CREATE INDEX indextable1 on table maintable(c) as 'carbondata'")
+    sql("INSERT INTO maintable SELECT 1,'string1', 'string2'")
+    sql("INSERT INTO maintable SELECT 1,'string1', 'string2'")
+    sql("INSERT INTO maintable SELECT 1,'string1', 'string2'")
+    sql("DELETE FROM TABLE INDEXTABLE1 WHERE SEGMENT.ID IN(0,1,2)")
+    sql("REINDEX INDEX TABLE indextable1 ON MAINTABLE WHERE SEGMENT.ID IN (0,1)")
```

Comment:
technically speaking reindex should not be allowed at this point, because the segment still exists(even if it is MFD), the purpose of reindex command is to reload missing segments which is not the case here. Also, will this same behaviour also happen for other segment status, say success or partially success? @akashrn5 can give some input maybe

Reply:
So please provide a better solution to solve this issue: main table segment (success), SI table segment (marked for delete). The solution should not be "run clean files for SI table first", because it is not better than current one.

Reply:
i think, at first, delete segment directly from indexTable should not be allowed. For second case, where main table contains SUCCESS segments and SI has MFD, during reindex, before loading data again, we can delete old files in that segment

Reply:
1. "i think, at first, delete segment directly from indexTable should not be allowed." This is unit test, we are simulating the issue(main table semgent success and SI table segment MFD), for real user, they don't need to manually delete SI segment in order to get the issue, they will face this issue directly in their daily running without any manuallly operation.
2. "For second case, where main table contains SUCCESS segments and SI has MFD, during reindex, before loading data again, we can delete old stale files in that segment" Yes, you provide a solution and i agree this will work, but again, the solution you provide is not as good as current one because current one you don't need to delete the stale files, reindex can also work.

Reply:
1. User can face this scenario, what i meant is, allowing delete directly on index Table is also a new issue, when user does it by mistake,  will make main table and SI not in sync. Then, user unneesarily has to perform Reindex. So, you can handle this scenario in this PR.
2. If you are gonna return only index Files, always merge file name will be null. So, you can use return List of index Files alone instead of Map from getIndexFiles method.

Reply:
1. User will not delete SI segment directly, the issue (main table segment success and SI segment MFD) is because carbon has so many bugs which will lead to this situation even user only just do the loading and compaction. And actually in production, we (maintainers) need to "delete directly on index table" just because we need to recovery carbon table from broken state to normal state, then later trigger reindex to repair the broken SI segment. So until "all the bugs which will lead to main table and SI table not in syc" are solved, we cannot block this.
2. This "return List of index Files alone" if you say so, I can change it.

Reply:
1. For example, if both main table and SI table segment status are success, but one of the carbondata file or index file in SI segment is missing or broken, how you fix this if you cannot manually delete this SI segment?

Reply:
ok

Reply:
Agree with @Indhumathi27. IMHO, we shouldn't allow delete segments on index table itself. And during repair index, if have segment with partial data, we should delete the segment completely(segment folder, segment file, tablestatus entry for the segment) before proceeding with segment repair.

Reply:
1. "we shouldn't allow delete segments on index table itself." please refer to the second last comment right before your comment.
2. "And during repair index, if have segment with partial data, we should delete the segment completely(segment folder, segment file, probably tablestatus entry for the segment as well) before proceeding with segment repair." You suggestion of course is right but too more complicate comparing to existing implementation, so please first do the complete analysis and design and then we can discuss and plan next step.